app/src/main/java/org/mozilla/tv/firefox/ScreenController.kt

```diff
@@ -53,53 +79,114 @@ class ScreenController {
-        // correctly with multiple sessions).
-        val browserFragment = fragmentManager.findFragmentByTag(WebRenderFragment.FRAGMENT_TAG) as? WebRenderFragment
-        if (browserFragment != null && browserFragment.isVisible) {
-            // We can't call loadUrl on the Fragment until the view hierarchy is inflated so we check
-            // for visibility in addition to existence.
-            browserFragment.loadUrl(url)
+    // TODO: handle about:home and close overlay by button press separately
+    fun showNavigationOverlay(fragmentManager: FragmentManager?, toShow: Boolean) {
+        fragmentManager ?: return
+        var transaction = fragmentManager.beginTransaction()
+        val overlayFragment = fragmentManager.navigationOverlayFragment()
+        val renderFragment = fragmentManager.webRenderFragment()
+        if (toShow) {
+            stateMachine.overlayOpened()
+            transaction = transaction.show(overlayFragment)
+                // TODO note that hiding WebRenderFragment will not be possible under a split overlay
+                .hide(renderFragment)
         } else {
-            val session = Session(url, source = source)
-            context.webRenderComponents.sessionManager.add(session, selected = true)
+            stateMachine.overlayClosed()
+            transaction = transaction.hide(overlayFragment)
+                .show(renderFragment)
         }
+        transaction.commit()
     }
 
-    fun showPocketScreen(fragmentManager: FragmentManager) {
-        fragmentManager.beginTransaction()
-                .replace(R.id.container, PocketVideoFragment(), PocketVideoFragment.FRAGMENT_TAG)
-                .addToBackStack(null)
-                .commit()
+    fun handleBack(fragmentManager: FragmentManager): Boolean {
+        if (stateMachine.currentActiveScreen == ScreenControllerStateMachine.ActiveScreen.WEB_RENDER) {
+            val webRenderFragment = fragmentManager.webRenderFragment()
+            if (webRenderFragment.onBackPressed()) return true
+        }
+        return handleTransition(fragmentManager, stateMachine.backPress())
+    }
+
+    fun handleMenu(fragmentManager: FragmentManager) {
+        handleTransition(fragmentManager, stateMachine.menuPress())
+    }
+
+    private fun handleTransition(fragmentManager: FragmentManager, transition: ScreenControllerStateMachine.Transition): Boolean {
+        when (transition) {
+            ScreenControllerStateMachine.Transition.ADD_OVERLAY -> showNavigationOverlay(fragmentManager, true)
+            ScreenControllerStateMachine.Transition.REMOVE_OVERLAY -> showNavigationOverlay(fragmentManager, false)
+            ScreenControllerStateMachine.Transition.ADD_POCKET -> {
+                fragmentManager.beginTransaction()
+                    .show(fragmentManager.pocketFragment())
+                    .hide(fragmentManager.navigationOverlayFragment())
```

Comment:
The commit changing this line says, "Show before hide to handle default focus (e.g. settings)." What was the problem and why does this help? If it was important enough to change the code for, we should probably have a comment to prevent people from changing it back, right?